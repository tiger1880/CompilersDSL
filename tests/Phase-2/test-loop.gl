$ Loops


$ while loop
int x := 100

until (x > 0)
{
    until(x < 0)
    {
        a := "hello"
    }
    a := "bye"
}

until (x > 0)


{

}

until (i < 1){
    
    "hello from the outside"
    stop
}

$ for loop

repeat(int i := 0| i < n |i++){
    Tri(a, b, c)
    a == (1, 0) 
    advance

}

$ optional args

i := 1000
repeat(| i != n |i--)

{
    

}

repeat(int i := 0| |){
    
    i++
    i < 10
    stop
}


