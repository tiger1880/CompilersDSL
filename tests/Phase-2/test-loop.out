LINE COMMENT: $ Loops
LINE COMMENT: $ while loop
DATATYPE: int
IDENTIFIER: x
EQUAL: :=
INTEGERS: 100
WHILE: until
PARENTHESIS: (
IDENTIFIER: x
CMP_OP/BRACKET: >
INTEGERS: 0
PARENTHESIS: )
PARENTHESIS: {
WHILE: until
PARENTHESIS: (
IDENTIFIER: x
CMP_OP/BRACKET: <
INTEGERS: 0
PARENTHESIS: )
PARENTHESIS: {
IDENTIFIER: a
EQUAL: :=
STRING_LITERAL: "hello"
PARENTHESIS: }
IDENTIFIER: a
EQUAL: :=
STRING_LITERAL: "bye"
PARENTHESIS: }
WHILE: until
PARENTHESIS: (
IDENTIFIER: x
CMP_OP/BRACKET: >
INTEGERS: 0
PARENTHESIS: )
PARENTHESIS: {
PARENTHESIS: }
WHILE: until
PARENTHESIS: (
IDENTIFIER: i
CMP_OP/BRACKET: <
INTEGERS: 1
PARENTHESIS: )
PARENTHESIS: {
STRING_LITERAL: "hello from the outside"
STOP: stop
PARENTHESIS: }
LINE COMMENT: $ for loop
FOR: repeat
PARENTHESIS: (
DATATYPE: int
IDENTIFIER: i
EQUAL: :=
INTEGERS: 0
SEPERATOR: |
IDENTIFIER: i
CMP_OP/BRACKET: <
IDENTIFIER: n
SEPERATOR: |
IDENTIFIER: i
UNARY: ++
PARENTHESIS: )
PARENTHESIS: {
CONSTRUCTOR: Tri
PARENTHESIS: (
IDENTIFIER: a
PUNCTUATION: ,
IDENTIFIER: b
PUNCTUATION: ,
IDENTIFIER: c
PARENTHESIS: )
IDENTIFIER: a
EQ_CMP_OP: ==
PARENTHESIS: (
INTEGERS: 1
PUNCTUATION: ,
INTEGERS: 0
PARENTHESIS: )
CONTINUE: advance
PARENTHESIS: }
LINE COMMENT: $ optional args
IDENTIFIER: i
EQUAL: :=
INTEGERS: 1000
FOR: repeat
PARENTHESIS: (
SEPERATOR: |
IDENTIFIER: i
EQ_CMP_OP: !=
IDENTIFIER: n
SEPERATOR: |
IDENTIFIER: i
UNARY: --
PARENTHESIS: )
PARENTHESIS: {
PARENTHESIS: }
FOR: repeat
PARENTHESIS: (
DATATYPE: int
IDENTIFIER: i
EQUAL: :=
INTEGERS: 0
SEPERATOR: |
SEPERATOR: |
PARENTHESIS: )
PARENTHESIS: {
IDENTIFIER: i
UNARY: ++
IDENTIFIER: i
CMP_OP/BRACKET: <
INTEGERS: 10
STOP: stop
PARENTHESIS: }
